app.rb

```diff
@@ -8,18 +9,42 @@
 require_relative './lib/ui/io.rb'
 
 def initializer(size, token1, token2)
+  state = {is_playing: true, stop_playing: false, is_won: false}
```

Comment:
One last thing....

You decided to use a hash for this. Why did you choose that vs. creating a new class? Is this going to be easy or hard to change if you need to change the structure of the `state`?

Reply:
Sometimes if something is rather small, I wonder if I should make a class for it or not.  Class would be easier for changes and hiding the implementation.  I use a struct in the restart-replay branch, I think i will change it to a class there, create some methods for changing the state, and link it back here when done.